Content.Shared/StatusEffect/StatusEffectsSystem.cs

```diff
@@ -163,14 +166,27 @@ private void OnHandleState(EntityUid uid, StatusEffectsComponent component, ref
 
             (TimeSpan, TimeSpan) cooldown = (_gameTiming.CurTime, _gameTiming.CurTime + time);
 
-            // If they already have this status effect, add the time onto it's cooldown rather than anything else.
             if (HasStatusEffect(uid, key, status))
             {
-                status.ActiveEffects[key].Cooldown.Item2 += time;
+                status.ActiveEffects[key].CooldownRefresh = refresh;
+                if(status.ActiveEffects[key].CooldownRefresh)
```

Comment:
why bother storing it on the status effect state? just check `if(refresh)`

Reply:
I save the state in case it comes in handy in the future for some kind of checks. But sure, I can remove this, and check refresh directly.

Reply:
I don't see why anything would ever need to know whether the most recent application of an effect was cumulative or a refresh. So it probably just doesn't even need to be stored in the first place.